alacritty_terminal/src/selection.rs

```diff
@@ -50,6 +50,36 @@ impl<L> SelectionRange<L> {
             && (self.start.col <= col || (self.start.line != line && !self.is_block))
             && (self.end.col >= col || (self.end.line != line && !self.is_block))
     }
+
+    pub fn is_above_buffer_line(&self, line: L, num_lines: L) -> bool
+    where
+        L: PartialEq + PartialOrd,
+    {
+        self.start.line > line || (self.end.line > line && self.end.line < num_lines)
```

Comment:
Why does it matter if self.end.line is smaller than num_lines?

Reply:
To not clear selection which is out side of viewport, since you're technically not touching it?